src/compiler/global.d.ts

```diff
@@ -0,0 +1,10 @@
+export {};
+
+declare global {
+  // Primary export
+  var compile: typeof import('./compile').compile;
+
+  // Needed for worker-dom
+  var self: Window & typeof globalThis;
+  var Node: typeof window.Node;
```

Comment:
I like that this is here now, it's good self-documentation of expected global state, nice change